src/App.tsx

```diff
@@ -1,26 +1,215 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { addTodo, deleteTodo, getTodos, updateTodo } from './api/todos';
+import { Errors } from './types/Errors';
+import { FilterTodosBy } from './types/FilterTodosBy';
+import { Error } from './components/Error/Error';
+import { filterTodos } from './utils/utils';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(Errors.NoError);
+  const [filterBy, setFilterBy] = useState(FilterTodosBy.All);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [updatingTodoId, setUpdatingTodoId] = useState<number | null>(null);
+  const [toggleCompleteAll, setToggleCompleteAll] = useState(false);
+  const [isHeaderDisabled, setIsHeaderDisabled] = useState(false);
+  const [tempIdCounter, setTempIdCounter] = useState(0);
+  const [deletingCardId, setDeletingCardId] = useState<number | null>(null);
+
+  useEffect(() => {
+    setHasError(Errors.NoError);
+
+    (async () => {
+      try {
+        const response = await getTodos();
+
+        setTodos(response);
+      } catch {
+        setHasError(Errors.UnableToLoad);
+      }
+    })();
```

Comment:
i think it will be more readable
```suggestion
    setHasError(Errors.NoError);
    
    const getAllTodos = async () => {
      try {
        const response = await getTodos();

        setTodos(response);
      } catch {
        setHasError(Errors.UnableToLoad);
      }
    }
    
    getAllTodos();
```